aiida/daemon/runner.py

```diff
@@ -29,15 +30,28 @@ def start_daemon():
     configure_logging(daemon=True, daemon_log_file=daemon_client.daemon_log_file)
 
     runner = DaemonRunner(rmq_config=get_rmq_config(), rmq_submit=False)
-    set_runner(runner)
 
+    def shutdown_daemon(num, frame):
+        logger.info('Received signal to shut down the daemon runner')
+        runner.close()
+        runner.stop()
+
+    signal.signal(signal.SIGINT, shutdown_daemon)
+    signal.signal(signal.SIGTERM, shutdown_daemon)
+
+    logger.info('Starting a daemon runner')
+
+    set_runner(runner)
     tick_legacy_workflows(runner)
 
     try:
         runner.start()
-    except (SystemError, KeyboardInterrupt):
-        logger.warning('Shutting down daemon')
+    except SystemError as exception:
```

Comment:
I'm not sure if we really need that since `SystemError` is meant to be used for internal Python interpreter errors, but there's no harm in keeping it.